### ISSUE TITLE
Name and state is not defined for ‘Add Font’ icon button present in ‘Style’ tab.

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#108]
User Experience:

User would find difficulty to identify the controls if name and state are not defined for any control as they won't get the context of that control and also they won't be able to perform functionality  relevant to that control. 

 

Note: User credentials should NOT be included in the bug.

Repro-steps:

1. Open Azure portal with valid credentials and create a API Management.
2. Click Developer Portal button to navigate to the portal.
3. Wait portal to load in design mode
4. Navigate to the Left nav and select "Styles" button.
5. Navigate to the various controls of the "Styles" window.

6. Open developer tool using F12 key and navigate to ‘Add Font’ button.

7. Verify the issue.

 

Actual Result:

Name and state are not defined for ‘Add Font’ icon button present in ‘Style’ tab, screen reader announces only ‘Button’

 

Expected Result:

Name should be defined as ‘Add Font’ button present in ‘Style’ tab, screen reader should announce as ‘Add font button’ and current state as expended ,collapse should also be defined.

 

Note:

Similar issue exist with Chrome.

Similar issue exist with button controls : ‘Add variation, Add color, Add shadow’ present in same page.

Similar issue exist throughout the application.

 

MAS Reference:

MAS 4.1.2 - Name, Role, Value

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
